docker/build_image.sh

```diff
@@ -130,17 +131,31 @@ then
   DOCKER_TAG="pytorch/torchserve:codebuild-$MACHINE"
 fi
 
+if [ "${BUILD_TYPE}" == "ci" ] && ! $USE_CUSTOM_TAG ;
+then
+  DOCKER_TAG="pytorch/torchserve:ci-$MACHINE"
+fi
+
 if [ "$USE_CUSTOM_TAG" = true ]
 then
   DOCKER_TAG=${CUSTOM_TAG}
 fi
 
+# Build docker base image
+DOCKER_BUILDKIT=1 docker build --file Dockerfile.base --build-arg BASE_IMAGE="${BASE_IMAGE}" --build-arg CUDA_VERSION="${CUDA_VERSION}"  --build-arg PYTHON_VERSION="${PYTHON_VERSION}" -t $DOCKER_BASE_IMAGE .
+
 if [ "${BUILD_TYPE}" == "production" ]
 then
   DOCKER_BUILDKIT=1 docker build --file Dockerfile --build-arg BASE_IMAGE="${BASE_IMAGE}" --build-arg CUDA_VERSION="${CUDA_VERSION}"  --build-arg PYTHON_VERSION="${PYTHON_VERSION}" -t "${DOCKER_TAG}" .
 elif [ "${BUILD_TYPE}" == "benchmark" ]
 then
   DOCKER_BUILDKIT=1 docker build --pull --no-cache --file Dockerfile.benchmark --build-arg USE_LOCAL_SERVE_FOLDER=$USE_LOCAL_SERVE_FOLDER --build-arg BASE_IMAGE="${BASE_IMAGE}" --build-arg BRANCH_NAME="${BRANCH_NAME}" --build-arg CUDA_VERSION="${CUDA_VERSION}" --build-arg MACHINE_TYPE="${MACHINE}" --build-arg PYTHON_VERSION="${PYTHON_VERSION}" -t "${DOCKER_TAG}" .
+elif [ "${BUILD_TYPE}" == "ci" ]
```

Comment:
How is ci different from dev? Can't we just have those be the same?

Reply:
dev will have torchserve starting when container is run. 
ci is running the regression tests script on start

Reply:
hmm, can we at least have the root of both dev and regression images be the same? I.e. having a common dev base and then just add the expose ports and start torchserve separatelty?
Speaking of this. What is the intention of the docker regression test? For me it is to test the functionality of TS inside the docker files we deploy? In that case we should stay as close to the actual images we deploy as possible. Ideally using the actually deployed images as  base and adding onmly stuff needed for the regression test on top to avoid diversion over time.

Reply:
For me it should loo like
```
Dockerfile -> Docker.run_time
         L -> Docker.dev -> Docker.regression (of use run_time as base for this)
```

Reply:
We can. I didn't do it yet since we plan to release next week. If we can push the release by a few weeks, I can test various combinations.